src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -2521,18 +2521,37 @@ pbs_python_populate_svrattrl_from_python_class(PyObject *py_instance,
 	PyObject	*py_resc_hookset_dict0 = (PyObject *)NULL;
 	PyObject	*py_attr_keys = (PyObject *)NULL;
 	PyObject 	*py_val = (PyObject *)NULL;
-	char	*name_str_dup = NULL;
-	char	*val_str_dup = NULL;
+	char		*name_str_dup = NULL;
+	char		*val_str_dup = NULL;
 	int		num_attrs, i;
 	pbs_list_head	svrattrl_list2;
-	int         rc = -1;
+	int		rc = -1;
 	int		hook_set_flag = 0;
 	int		has_resv_duration;
-	char            the_resc[HOOK_BUF_SIZE];
-	char            the_val[HOOK_BUF_SIZE];
+	static char     *the_resc = NULL;
+	static char     *the_val = NULL;
+	static long int resc_buf_size = HOOK_BUF_SIZE;
+	static long int val_buf_size = HOOK_BUF_SIZE;
```

Comment:
addition of static variables makes the function non re-entrant. I would suggest that you study this function and find if it is already non re-entrant or not. If yes, add a note in the doxygen.
If not, we should remove these static variables and find other way to solve the problem at hand.